apps/src/blockly/addons/cdoXml.js

```diff
@@ -118,3 +129,12 @@ export default function initializeBlocklyXml(blocklyWrapper) {
   // We don't want to save absolute position in the block XML
   blocklyWrapper.Xml.blockToDomWithXY = blocklyWrapper.Xml.blockToDom;
 }
+
+// Compare function - Moves functional definitions to the end of a block list.
+function reorderBlocks(x, y) {
```

Comment:
Why do we need to reorder if y is a function block? I would have thought this would be if x is function block, reorder, otherwise do nothing?

Reply:
This function can be read like, "if x is a function block, sort it after y. if y is a function block, sort x before y".

I used the table here to plan this function: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort#description

Reply:
Cool, LGTM! I don't find double ternarys super readable but just personal preference, something like this easier for me to read (but verbose):

https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/achievements.js#L9-L17

Reply:
Does this seem easier to read? [b11f7fe](https://github.com/code-dot-org/code-dot-org/pull/48631/commits/b11f7feb4fe2dc66cc5225bff9e2a96184195557)
I'm going to try it as an if/else-if/else and see if that's even better.